modules/internal/ChapelBase.chpl

```diff
@@ -1286,23 +1306,29 @@ module ChapelBase {
     __primitive("/=", lhs, rhs);
   }
   inline proc /=(ref lhs:real(?w), rhs:real(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0.0 then
+        halt("Attempt to divide by zero");
```

Comment:
I don't believe we want divide by zero checks for reals because floating point semantics should handle it without a C-level exception.  Note that we don't support them for the plain-old '/' operator, for example.